### ISSUE TITLE
[Starfish] Add db attributes to database span's span data

### ISSUE THREAD

**Issue by AbhiPrasad** [getsentry/sentry-java#2893]
To get an overview of this work across all SDKs, please visit:
https://github.com/orgs/getsentry/projects/135

Parent issue: https://github.com/getsentry/team-sdks/issues/19

For the starfish db module it was determined that we need extra database attributes, specifically about the database connection attributes.

**NOTE: this should apply to both android and java db spans**

We're going to matching the [otel db conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/database/). Either one of `server.address` or `server.socket.address` is required. If `server.address` is set, then `server.port` should be set. If `server.port` is different than `server.socket.port`, than `server.socket.port` should also be set. In addition, `db.name` should be set if possible. `db.system` should match the list in [OpenTelemetry's well known conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#notes-and-well-known-identifiers-for-dbsystem).

**The main priority items are db.name and db.system**

| Attribute  | Type | Description  | Examples  | Requirement Level |
|---|---|---|---|---|
| `db.system` | string | An identifier for the database management system (DBMS) product being used. | `mysql` | Required |
| `server.address` | string | Name of the database host. | `example.com` | Conditionally Required: See alternative attributes below. |
| `server.port` | int | Logical server port number | `80`; `8080`; `443` | Conditionally Required: [1] |
| `server.socket.address` | string | Physical server IP address or Unix socket address. | `10.5.3.2` | See below |
| `server.socket.port` | int | Physical server port. | `16456` | Recommended: If different than `server.port`. |
| `db.name` | string | This attribute is used to report the name of the database being accessed. For commands that switch the database, this should be set to the target database (even if the command fails). | `customers`; `main` | Conditionally Required: If applicable. |

These values should be set on `span.data`.

**NOTE: For the Mobile SDKs only `db.name` and `db.system` is required**

### Motivation

This is being done for the Sentry's [new performance views for databases](https://sentry.sentry.io/starfish/database/). See a WIP example of the view below:

<img width="1251" alt="image" src="https://github.com/getsentry/sentry-laravel/assets/18689448/962c5fea-1b38-481c-ba22-7046a33ea454">

**Comment by adinauer** [comment]
`db.system` and `db.name` were added via #2894 as a first step. Other info has to be parsed from JDBC connection string as well.

**Comment by romtsn** [comment]
On Android we should check whether it's possbile:
- To distinguish between in-memory and sqlite for `db.system` (if not, we should probably always send `sqlite`)
- To retrieve the database name 

@AbhiPrasad we also instrument Room (ORM) spans, but they don't have the underlying SQL query as a description in them. Would it make sense to add these attributes to the ORM spans at all? The ORM spans now have span op `db` - should we perhaps change that to something more specific (`db.sql.room`) to not be accounted for in starfish?

This is how it looks right now 
![image](https://github.com/getsentry/sentry-java/assets/4999776/371b9674-0b6e-4183-908a-e5cfa16928df)

**Comment by AbhiPrasad** [comment]
We need to make sure the starfish views account for the different types of db spans - we'll take care of that. If we could make them their own special span op that would help quite a bit.

> Would it make sense to add these attributes to the ORM spans at all

I think it still makes sense to add `db.name` and `db.system`, gives us room to build views for that data going forward.

**Comment by romtsn** [comment]
@AbhiPrasad is it fine to close this issue once the Android PR is merged, or do we wanna keep it for the java-backend missing things (like `server.{stuff}`)? We could also create a follow-up issue to not have this one hanging around forever

**Comment by AbhiPrasad** [comment]
> is it fine to close this issue once the Android PR is merged

Yes that sounds good to me. Creating follow-up issues for the Java server specific stuff sounds like a good idea to me.
